src/Microsoft.Health.Blob/Features/Storage/BlobClientFactory.cs

```diff
@@ -26,7 +26,7 @@ public static BlobServiceClient Create(BlobDataStoreConfiguration configuration)
 
             if (configuration.AuthenticationType == BlobDataStoreAuthenticationType.ManagedIdentity)
             {
-                DefaultAzureCredential credential = new DefaultAzureCredential(new DefaultAzureCredentialOptions { ManagedIdentityClientId = configuration.ManagedIdentityClientId });
+                DefaultAzureCredential credential = new DefaultAzureCredential(configuration.Credentials);
```

Comment:
Should we always use these credentials, regardless of managed identity?

Reply:
Looks like BlobServiceClient assumes that a string connectionstring will contain its own credentials. It doesn't give the option to use string + credentials in the constructor https://docs.microsoft.com/en-us/dotnet/api/azure.storage.blobs.blobserviceclient.-ctor?view=azure-dotnet#azure-storage-blobs-blobserviceclient-ctor(system-string-azure-storage-blobs-blobclientoptions)

Reply:
Oh you know, that makes perfect sense.